crates/routes/src/image_proxy.rs

```diff
@@ -0,0 +1,34 @@
+use actix_web::{
+  web,
+  web::{Query, ServiceConfig},
+  HttpResponse,
+};
+use lemmy_api_common::context::LemmyContext;
+use lemmy_utils::{error::LemmyResult, rate_limit::RateLimitCell};
+use serde::Deserialize;
+use urlencoding::decode;
+
+pub fn config(cfg: &mut ServiceConfig, rate_limit: &RateLimitCell) {
+  cfg.service(
+    web::resource("/api/v3/image_proxy")
+      .wrap(rate_limit.message())
+      .route(web::post().to(image_proxy)),
+  );
+}
+
+#[derive(Deserialize)]
+struct ImageProxyParams {
+  url: String,
+}
+
+async fn image_proxy(
```

Comment:
@asonix would using the lemmy-backend to proxy picture requests be a better approach, or maybe waiting for pictrs v.50 which has picture proxying, and using those pictrs routes? Any pros/cons to either?

Reply:
proxying images through lemmy is fine, but it means the original server must be online and capable of serving the image for each request. using pict-rs' proxy method in 0.5 will cache the proxied image and reduce load on the original server 

I think doing this initial proxy work now probably makes sense, and using pict-rs' proxying in the future can be made a lower priority (and give more time to folks to upgrade to 0.5 when it releases before you start depending on 0.5 endpoints)

Reply:
In that case @Nutomic just add a TODO somewhere that we can remove this once pictrs `0.5` is released.

Reply:
Sounds good, adding comment.

Reply:
Changed it to use pictrs for proxying. Also added a config option which is disabled by default, with comment noting that pictrs 0.5 is required.